trace/src/normalization/normalizer.rs

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0. This product includes software
+// developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present
+// Datadog, Inc.
+
+use std::time::{SystemTime};
+use crate::errors;
+use crate::normalize_utils;
+use crate::pb;
+
+const MAX_TYPE_LEN: i64 = 100;
+
+// an arbitrary cutoff to spot weird-looking values
+// nanoseconds since epoch on Jan 1, 2000
+const YEAR_2000_NANOSEC_TS: i64 = 946684800000000000;
+
+#[allow(dead_code)]
+fn normalize(s: &mut pb::Span) -> Result<(), errors::NormalizerError> {
+    if s.trace_id == 0 {
+        return Err(errors::NormalizerError::new("TraceID is zero (reason:trace_id_zero)"));
+    }
+    if s.span_id == 0 {
+        return Err(errors::NormalizerError::new("SpanID is zero (reason:span_id_zero)"));
+    }
+    
+    // TODO: Implement service name normalizer in future PR
+    // let (svc, err) = normalize_utils::normalize_service(s.service.clone(), "".to_string());
+    // match err {
+    //     Some(errors::NormalizeErrors::ErrorEmpty) => println!("Fixing malformed trace. Service is empty (reason:service_empty)"),
+    //     Some(errors::NormalizeErrors::ErrorTooLong) => println!("Fixing malformed trace. Service is too long (reason:service_truncate)"),
+    //     Some(errors::NormalizeErrors::ErrorInvalid) => println!("Fixing malformed trace. Service is invalid (reason:service_invalid)"),
+    //     None => ()
+    // }
+    // s.service = svc;
+
+    // TODO: check for a feature flag to determine the component tag to become the span name
+    // https://github.com/DataDog/datadog-agent/blob/dc88d14851354cada1d15265220a39dce8840dcc/pkg/trace/agent/normalizer.go#L64
+
+    let (normalized_name, err) = normalize_utils::normalize_name(s.name.clone());
+    match err {
+        Some(errors::NormalizeErrors::ErrorEmpty) => println!("Fixing malformed trace. Name is empty (reason:span_name_empty)"),
+        Some(errors::NormalizeErrors::ErrorTooLong) => println!("Fixing malformed trace. Name is too long (reason:span_name_truncate)"),
+        Some(errors::NormalizeErrors::ErrorInvalid) => println!("Fixing malformed trace. Name is invalid (reason:span_name_invalid)"),
+        None => ()
+    }
+    s.name = normalized_name;
+
+    if s.resource.is_empty() {
+        println!("Fixing malformed trace. Resource is empty (reason:resource_empty)");
```

Comment:
I'm guessing this should be a log message instead, but we don't have a logging story in libdatadog yet. At least, this shouldn't print to stdout. The same goes for the other prints.

Reply:
Right, it should be. I'll remove all the print statements